modules/core/src/main/java/org/apache/ignite/internal/managers/discovery/ClusterMetricsImpl.java

```diff
@@ -350,15 +364,23 @@ public ClusterMetricsImpl(GridKernalContext ctx, GridLocalMetrics vmMetrics, lon
 
     /** {@inheritDoc} */
     @Override public int getTotalNodes() {
-        return 1;
+        IntMetricImpl totalNodes = sysreg.intMetric("TotalNodes", null);
+        totalNodes.add(1);
+
+        return totalNodes.value();
     }
 
     /** {@inheritDoc} */
     @Override public long getCurrentPmeDuration() {
         GridDhtPartitionsExchangeFuture future = ctx.cache().context().exchange().lastTopologyFuture();
-        
-        return (future == null || future.isDone()) ?
+
```

Comment:
Seems, we should change `PmeStartTime` on `GridDhtPartitionsExchangeFuture` start instead of calculation on query.